common/lib/protocol-definitions/src/summary.ts

```diff
@@ -21,41 +21,88 @@ export interface ISummaryCommitter {
     date: string;
 }
 
+/**
+ *  Represents a node from the Summary Tree.
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
 export namespace SummaryType {
     export type Tree = 1;
     export type Blob = 2;
     export type Handle = 3;
     export type Attachment = 4;
 
-    export const Tree: Tree = 1 as const;
-    export const Blob: Blob = 2 as const;
-    export const Handle: Handle = 3 as const;
-    export const Attachment: Attachment = 4 as const;
+    /**
+     *  Represents a sub-tree in the summary.
+     */
+     export const Tree: Tree = 1 as const;
+
+     /**
+      * Represents a blob of data that is added to the summary.
+      * Such as the user data that is added to the DDS or metadata added by runtime
+      * such as data store / channel attributes.
+      */
+     export const Blob: Blob = 2 as const;
+
+     /**
+      * Path to a summary tree object from the last successful summary.
+      */
+     export const Handle: Handle = 3 as const;
+
+     /**
+      * Unique identifier to larger blobs uploaded outside of the summary.
+      * Ex. DDS has large images or video that will be uploaded by the BlobManager and
+      * receive an Id that can be used in the summary.
+      */
+     export const Attachment: Attachment = 4 as const;
 }
 export type SummaryType = SummaryType.Attachment | SummaryType.Blob | SummaryType.Handle | SummaryType.Tree;
 
 export type SummaryTypeNoHandle = SummaryType.Tree | SummaryType.Blob | SummaryType.Attachment;
 
+/**
+ * Path to a summary tree object from the last successful summary indicating the summary object hasn't
+ * changed since it was uploaded.
+ * To illustrate, if a DataStore did not change since last summary, the framework runtime will use a handle for the
+ * entire DataStore instead of re-sending the entire subtree. The same concept applies for a DDS.
+ * An example of handle would be: '/<DataStoreId>/<DDSId>'.
+ * Notice that handles are optimizations from the Fluid Framework Runtime and the DDS is not aware of the handles.
+ */
 export interface ISummaryHandle {
     type: SummaryType.Handle;
 
-    // No handles, all other SummaryType are Ok
+    /**
+     * Type of Summary Handle (SummaryType.Handle is not supported).
+     */
     handleType: SummaryTypeNoHandle;
 
-    // Stored handle reference
+    /**
+     * Unique path that identifies the stored handle reference.
+     */
     handle: string;
 }
 
+/**
```

Comment:
I think it would be good to mention here that to reference an already uploaded blob use ISummaryAttachment. Someone looking for how to do that is likley to find themselves here, and not realize they need to look into ISummaryAttachments.